Nasal/engine.nas

```diff
@@ -88,6 +88,55 @@ var primerTimer = maketimer(5, func {
     primerTimer.stop();
 });
 
+# ========== oil consumption ======================
+
+var oil_consumption = maketimer(1.0, func {
+    var oil_level = getprop("/engines/active-engine/oil-level");
+    var rpm = getprop("/engines/active-engine/rpm");
+    # quadratic formula which outputs 1.0 for input 2300 RPM (cruise value), 0.6 for 700 RPM (idle) and 1.2 for 2700 RPM (max)
+    var rpm_factor = 0.00000012 * math.pow(rpm,2) - 0.0001 * rpm + 0.62;
+    # consumption rate defined as 1.5 quarter per 10 hours (36000 seconds) at cruise RPM
+    var consumption_rate = 1.5 / 36000; 
+    var low_oil_pressure_factor = 1.0;
+    var low_oil_temperature_factor = 1.0;
+    
+    if ((getprop("/engines/active-engine/running")) and (getprop("/engines/active-engine/oil_consumption_allowed"))) {
+        oil_level = oil_level - consumption_rate * rpm_factor;
+        setprop("/engines/active-engine/oil-level", oil_level);        
+    }
+    
+    # if oil gets low, pressure should lower and temperature should rise
+    if (oil_level < 5.0) {
+        low_oil_pressure_factor = 11.25 * oil_level - 55.25; # should give 1.0 for oil_level = 5 and 0.1 for oil_level 4.92, which is the min before the engine stops
```

Comment:
Also, try to put comments above the code instead of after the code to prevent long lines. Normally you keep code and/or comments limited to 72 to 79 characters.